File: car_factory/src/features/chat/chat-api.ts
import { httpsCallable } from "firebase/functions";
import { getClientFunctions, isFirebaseConfigured } from "@/lib/firebase/client";

export async function getOrCreateChatRoom(
  productId: string,
  options?: { buyerUid?: string },
) {
  if (!isFirebaseConfigured()) {
    throw new Error("Firebase가 설정되지 않았습니다.");
  }
  const fn = httpsCallable<
    { productId: string; buyerUid?: string },
    { roomId: string }
  >(getClientFunctions(), "getOrCreateChatRoom");
  const result = await fn({
    productId,
    ...(options?.buyerUid ? { buyerUid: options.buyerUid } : {}),
  });
  const roomId = result.data?.roomId?.trim();
  if (!roomId) throw new Error("채팅방을 만들지 못했습니다.");
  return roomId;
}

export async function markChatRoomRead(roomId: string) {
  if (!isFirebaseConfigured()) return;
  const fn = httpsCallable<{ roomId: string }, { ok: boolean }>(
    getClientFunctions(),
    "markChatRoomRead",
  );
  await fn({ roomId });
}

File: car_factory_app/lib/features/chat/chat_api.dart
import 'package:cloud_functions/cloud_functions.dart';

import '../../core/firebase/firebase_bootstrap.dart';

/// 채팅방 — 웹 `getOrCreateChatRoom` Callable 대응.
class ChatApi {
  ChatApi({FirebaseFunctions? functions})
      : _functions = functions ?? FirebaseBootstrap.functions;

  final FirebaseFunctions _functions;

  Future<String> getOrCreateChatRoom({required String productId}) async {
    final callable = _functions.httpsCallable('getOrCreateChatRoom');
    final result = await callable.call<Map<String, dynamic>>({
      'productId': productId,
    });
    final data = Map<String, dynamic>.from(result.data as Map);
    final roomId = data['roomId'] as String?;
    if (roomId == null || roomId.isEmpty) {
      throw StateError('채팅방을 만들지 못했습니다.');
    }
    return roomId;
  }

  Future<void> markChatRoomRead({required String roomId}) async {
    final callable = _functions.httpsCallable('markChatRoomRead');
    await callable.call({'roomId': roomId});
  }
}
